heuristic/local_search/reinsertion.py

```diff
@@ -0,0 +1,31 @@
+from numpy.random import RandomState
+
+from heuristic.classes import Solution, Problem
+from heuristic.repair_operators import greedy_insert
+
+
+def reinsertion(current: Solution, rnd_state: RandomState) -> Solution:
+    """
+    Attempts to find an improvement in the current solution by removing a
+    customer and reinserting this customer at the optimal location. If an
+    improvement is found the new solution is returned.
+    """
+    problem = Problem()
+
+    for customer in range(problem.num_customers):
+        destroyed = current.copy()
```

Comment:
This is done here, _and_ below in greedy_insert. Copying is really expensive, so I'd avoid this like the plague. If you can do something without copying that's a lot better.

Reply:
Greedy insert does not copy I believe?